src/Entity/Assignment.php

```diff
@@ -134,12 +137,45 @@ public function isCancellable(): bool
         return in_array($this->state, [self::STATE_STARTED, self::STATE_READY], true);
     }
 
+    public function getAttemptsCount(): int
+    {
+        return $this->attemptsCount;
+    }
+
+    public function setAttemptsCount(int $attemptsCount): self
+    {
+        $this->attemptsCount = $attemptsCount;
+
+        return $this;
+    }
+
+    public function incrementAttemptsCount(): self
+    {
+        $this->attemptsCount++;
+
```

Comment:
Shouldn't we have a check somewhere upon calling this? Mean that it won't be the case that `attemptsCount` would be incremented to the value > than maxAttampts.
Or there is already such a check, I'm just blind? :upside_down_face:

Reply:
You're right it was missing, fixed it :) !